qiling/os/os.py

```diff
@@ -304,18 +304,18 @@ def _call_api(self, name, params, result, address, return_address):
 
     def x86_stdcall(self, param_num, params, func, args, kwargs, passthru=False):
         # if we check ret_addr before the call, we can't modify the ret_addr from inside the hook
-        result, param_num = self.__x86_cc(param_num, params, func, args, kwargs)
+        result, param_num = self.__x86_cc(param_num, params, func, args, kwargs, passthru)
 
         # get ret addr
         ret_addr = self.ql.stack_read(0)
 
         # append syscall to list
         self._call_api(func.__name__, params, result, self.ql.reg.arch_pc, ret_addr)
 
-        # update stack pointer
-        self.ql.reg.arch_sp = self.ql.reg.arch_sp + ((param_num + 1) * 4)
-
         if not passthru and self.PE_RUN:
+            # update stack pointer
+            self.ql.reg.arch_sp = self.ql.reg.arch_sp + ((param_num + 1) * 4)
```

Comment:
If the original binary is handling the implementation (passthru), the original function should also clean up the stack.